src/seedu/addressbook/data/person/PostalCode.java

```diff
@@ -0,0 +1,108 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.net.URLConnection;
+
+/**
+ * Represents an PostalCode's postal code in the PostalCode book.
+ * Guarantees: immutable; is valid as declared in {@link #isValidPostalCode(String)}
+ */
+public class PostalCode {
+
+    public static final String EXAMPLE = "119077";
+    public static final String MESSAGE_POSTAL_CODE_CONSTRAINTS = "Postal codes should be in a 6-digit format";
+    public static final String POSTAL_CODE_VALIDATION_REGEX = "\\d{6}";
+    public static final int NUM_CHARS_AFTER_ADDRESS_SUBSTRING = 10;
+    public static final int NUM_CHARS_BEFORE_POSTAL_SUBSTRING = 3;
+    public static final String MESSAGE_ERROR_ADDRESS_NOT_FOUND = "Address not found.";
+    public static final String ONEMAP_SEARCH_URL_FIRST_HALF = "https://developers.onemap.sg/commonapi/search?searchVal=";
+    public static final String ONEMAP_SEARCH_URL_SECOND_HALF = "&returnGeom=N&getAddrDetails=Y";
+
+    public final String value;
+    private boolean isPrivate;
+
+    /**
+     * Validates given PostalCode.
+     *
+     * @throws IllegalValueException if given PostalCode string is invalid.
+     */
+    public PostalCode(String postalCode, boolean isPrivate) throws IllegalValueException {
+        String trimmedPostalCode = postalCode.trim();
+        this.isPrivate = isPrivate;
+        if (!isValidPostalCode(trimmedPostalCode)) {
+            throw new IllegalValueException(MESSAGE_POSTAL_CODE_CONSTRAINTS);
+        }
+        this.value = trimmedPostalCode;
+    }
+
+    /**
+     * Returns true if a given string is a valid person PostalCode.
+     */
+    public static boolean isValidPostalCode(String test) {
+        return test.matches(POSTAL_CODE_VALIDATION_REGEX);
+    }
+
+    public String retrieveMatchingAddress() throws IOException {
+        try {
+            String lineFromURL = getOutputFromOnemapURL();
+            String address = getAddressFromUrlOutput(lineFromURL);
+            if (address != null) return address;
+        }
+        catch(IOException e) {
+            return MESSAGE_ERROR_ADDRESS_NOT_FOUND;
+        }
+        return MESSAGE_ERROR_ADDRESS_NOT_FOUND;
+    }
+
```

Comment:
Missing header comment. All non trivial methods should have javadoc format header comments.